NetworkRegistry/NetworkRegistry.md

```diff
@@ -0,0 +1,65 @@
+---
+CIP: TODO
+Title: Chain ID Registry
+Authors: Sebastien Guillemot <seba@dcspark.io>
+Comments-URI: TODO
+Status: Draft
+Type: Informational
+Created: 2021-11-24
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Currently Cardano has two easily-usable networks: "mainnet" and "testnet". However, in the future, we expect more networks to exist and so we need some way to refer to these networks to be able to write better multi-network applications and systems.
+
+## Terminology
+
+### Existing network IDs
+
+Cardano currently has three ways to refer to a network:
+- The "network ID" included in every address as part of [bech32](https://github.com/bitcoinjs/bech32). This only stores 16 possibilities (4 bits)
+- The "network magic" used for Byron addresses. This is a random 32-bit number
+- The genesis block hash (a 28-byte number)
+
+## Motivation
+
+Blockchains can have multiple deployments of the same codebase. For example:
+
+1. Test networks where the base asset has no value (so devs can test at no cost)
+1. Test networks where the protocol is simplified for ease of testing (ex: Cardano but with a Proof of Authority conensus for stable block production in testing)
+1. Test networks for new features
+    1. PAB testnet
+    2. Shelley incentivized testnet
+1. Forks that diverge in feature set (the many forks of Bitcoin and Ethereum)
+
+dApps may be deployed on specific testnets that match their criteria and wallets need to know about these networks to know how to behave.
+
+Additionally, having a standardized registry for networks allows easy integration into the broader crypto ecosystem via standards like [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md)
+
+## Specification
+
+We create a machine-readable registry of networks
+
+All entries in this registry should have the following entries:
+
+- User-friendly name
+- Network ID
+- Network magic
+- Genesis hash
+
+When representing these networks in a human-readable string, the following format shall be used:
+
+```
```

Comment:
Another option for the format could be bech32. The result would look something like this:
```
cipxx:1pugjyv6y3rn0jc
```

This is less human readable, but not that bad and has the checksum in it which is also nice (but the checksum is a little aggressive for the size of the payload. Checksum is 6 characters and so around half of the resulting string is just the checksum)